go/pserver/service_test.go

```diff
@@ -166,3 +168,7 @@ func TestBlockUntilInitialized(t *testing.T) {
 
 	wg.Wait()
 }
+
+func TestCheckpointSpeed(t *testing.T) {
```

Comment:
leave a TODO here, will be tested after reaching an agreement with @Yancey1989 's recover logic.